### PR TITLE
eth_feeHistory implementation

### PR DESCRIPTION
**Pull request by sunce86**
Added the eth_feeHistory function that was introduced as part of the London hard fork.

### CODE REVIEW

File: src/api/eth.rs [hunk 1]
```diff
@@ -88,6 +89,23 @@ impl<T: Transport> Eth<T> {
         CallFuture::new(self.transport.execute("eth_gasPrice", vec![]))
     }
 
+    /// Get fee history
+    pub fn fee_history(
+        &self,
+        block_count: U256,
+        oldest_block: BlockNumber,
```
Comment (tomusdrw) [inline src/api/eth.rs:96]:
```suggestion
        newest_block: BlockNumber,
```

File: src/api/eth.rs [hunk 2]
```diff
@@ -88,6 +89,23 @@ impl<T: Transport> Eth<T> {
...
+        reward: Option<Vec<f64>>,
```
Comment (tomusdrw) [inline src/api/eth.rs:97]:
```suggestion
        reward_percentiles: Option<Vec<f64>>,
```

File: src/types/block.rs [hunk 3]
```diff
@@ -158,6 +158,24 @@ impl Serialize for BlockNumber {
...
+            "earliest" => Ok(BlockNumber::Earliest),
+            "pending" => Ok(BlockNumber::Pending),
+            _ if value.starts_with("0x") => U64::from_str_radix(&value[2..], 16)
+                .map(BlockNumber::Number)
+                .map_err(|e| D::Error::custom(format!("invalid block number: {}", e))),
+            _ => Err(D::Error::custom("invalid block number: missing 0x prefix".to_string())),
+        }
+    }
+}
+
```
Comment (tomusdrw) [inline src/types/block.rs:178]: Could you add a test?

File: src/types/fee_history.rs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+use crate::types::{BlockNumber, U256};
+use serde::{Deserialize, Serialize};
+
+/// The fee history type returned from RPC calls.
```
Comment (tomusdrw) [inline src/types/fee_history.rs:4]:
```suggestion
/// The fee history type returned from `eth_feeHistory` call.
```

File: src/types/block.rs (hunk 3 above)
Comment (sunce86) [reply src/types/block.rs:178]: Added test.

File: src/types/fee_history.rs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            oldest_block: BlockNumber::Number(123456.into()),
+            base_fee_per_gas: vec![100.into(), 110.into()],
+            gas_used_ratio: vec![1.0, 2.0, 3.0],
+            reward: None,
+        };
+
+        let serialized = serde_json::to_value(fee_history.clone()).unwrap();
+        let deserialized = serde_json::from_value(serialized).unwrap();
+
+        assert_eq!(fee_history, deserialized);
```
[the remark this answers is not in the document]
Comment (sunce86) [reply src/types/fee_history.rs:35]: Added.

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**